Assets/Scripts/Models/Area/World.cs

```diff
@@ -242,14 +242,14 @@ public Tile GetFirstTileWithNoInventoryAround(int maxOffset, int centerX, int ce
             {
                 offsetY = offset;
                 tile = GetTileAt(centerX + offsetX, centerY + offsetY, centerZ);
-                if (tile.Inventory == null)
+                if (tile.Inventory == null && tile.Furniture == null)
```

Comment:
So, the TradingController does something that really should be the TradingModel that does. The tile placement shouldn't be figured out there, it should be a model thing. I've got in mind an upgrade that I'm going to work on that should resolve both. I'll leave the model logic in the TradingController for now, but...